recipes/ftjam/all/conanfile.py

```diff
@@ -20,29 +22,39 @@ def _source_subfolder(self):
         return "source_subfolder"
 
     def configure(self):
-        if self.settings.compiler == "Visual Studio":
-            raise ConanInvalidConfiguration("ftjam does not work, built with Visual Studio")
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("ftjam-{}".format(self.version), self._source_subfolder)
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("ftjam doesn't build with Visual Studio yet")
+        if hasattr(self, "settings_build") and tools.cross_building(self):
```

Comment:
Isn't `tools.cross_building(self)` sufficient?

Reply:
No, with 1 profile, `tools.cross_building(self)` is True on Linux machine evaluating package id when host is not Linux.

Reply:
Thanks.
Indeed, with 1 profile approach, it falls back on a comparison with `detected_os()` and `detected_architecture()`.